libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -175,28 +176,32 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 {
 	yulAssert(m_scope, "");
 
-	int const numVariables = _varDecl.variables.size();
-	int heightAtStart = m_assembly.stackHeight();
+	size_t const numVariables = _varDecl.variables.size();
+	auto heightAtStart = static_cast<size_t>(m_assembly.stackHeight());
 	if (_varDecl.value)
 	{
 		std::visit(*this, *_varDecl.value);
-		expectDeposit(numVariables, heightAtStart);
+		expectDeposit(static_cast<int>(numVariables), static_cast<int>(heightAtStart));
 	}
 	else
 	{
 		m_assembly.setSourceLocation(_varDecl.location);
-		int variablesLeft = numVariables;
+		size_t variablesLeft = numVariables;
 		while (variablesLeft--)
 			m_assembly.appendConstant(u256(0));
 	}
 
 	m_assembly.setSourceLocation(_varDecl.location);
 	bool atTopOfStack = true;
-	for (int varIndex = numVariables - 1; varIndex >= 0; --varIndex)
+	for (
+		size_t varIndex = 0, varIndexReverse = numVariables - 1 - varIndex;
+		varIndex < numVariables;
+		varIndexReverse = numVariables - 1 - ++varIndex
+	)
 	{
```

Comment:
That's pretty crafty. But I think this would be easier on the eyes:
```suggestion
	for (size_t varIndex = 0; varIndex < numVariables; ++varIndex)
	{
		size_t varIndexReverse = numVariables - 1 - varIndex;
```